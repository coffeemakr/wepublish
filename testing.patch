diff --git a/libs/api/src/lib/db/article.ts b/libs/api/src/lib/db/article.ts
index 7359af6a..ea27be09 100644
--- a/libs/api/src/lib/db/article.ts
+++ b/libs/api/src/lib/db/article.ts
@@ -14,11 +14,11 @@ export interface ArticleData {
   readonly lead?: string | null
   readonly seoTitle?: string | null
   readonly slug?: string | null
-  readonly tags: string[]
+  readonly tags?: string[]
 
   readonly canonicalUrl?: string | null
 
-  readonly properties: MetadataProperty[]
+  readonly properties?: MetadataProperty[]
 
   readonly imageID?: string | null
   readonly authors: ArticleRevisionAuthor[]
@@ -107,7 +107,7 @@ export type ArticleRevisionWithRelations = PrismaArticleRevision & {
 
 export type ArticleWithRevisions = PrismaArticle & {
   draft: ArticleRevisionWithRelations | null
-  pending: ArticleRevisionWithRelations | null
+  pending?: ArticleRevisionWithRelations | null
   published: ArticleRevisionWithRelations | null
 }
 
diff --git a/libs/api/src/lib/db/page.ts b/libs/api/src/lib/db/page.ts
index fb73c289..6f0ed1b5 100644
--- a/libs/api/src/lib/db/page.ts
+++ b/libs/api/src/lib/db/page.ts
@@ -14,9 +14,9 @@ export interface PageData {
 
   readonly title: string
   readonly description?: string | null
-  readonly tags: string[]
+  readonly tags?: string[]
 
-  readonly properties: MetadataProperty[]
+  readonly properties?: MetadataProperty[]
 
   readonly imageID?: string | null
 
@@ -85,7 +85,7 @@ export type PageRevisionWithProperties = PrismaPageRevision & {
 
 export type PageWithRevisions = PrismaPage & {
   draft: PageRevisionWithProperties | null
-  pending: PageRevisionWithProperties | null
+  pending?: PageRevisionWithProperties | null
   published: PageRevisionWithProperties | null
 }
 
diff --git a/libs/api/src/lib/graphql/article.ts b/libs/api/src/lib/graphql/article.ts
index 8d80fb27..77357e01 100644
--- a/libs/api/src/lib/graphql/article.ts
+++ b/libs/api/src/lib/graphql/article.ts
@@ -275,7 +275,7 @@ export const GraphQLPublicArticle: GraphQLObjectType<PublicArticle, Context> =
         type: GraphQLNonNull(GraphQLList(GraphQLNonNull(GraphQLMetadataPropertyPublic))),
         resolve: ({properties}) => {
           return properties
-            .filter(property => property.public)
+            ?.filter(property => property.public)
             .map(({key, value}) => ({key, value}))
         }
       },
diff --git a/libs/api/src/lib/graphql/article/article.private-mutation.ts b/libs/api/src/lib/graphql/article/article.private-mutation.ts
index 43a6a3ee..049c7e54 100644
--- a/libs/api/src/lib/graphql/article/article.private-mutation.ts
+++ b/libs/api/src/lib/graphql/article/article.private-mutation.ts
@@ -178,7 +178,8 @@ export const duplicateArticle = async (
       createMany: {
         data: socialMediaAuthors.map(({authorId}) => ({authorId}))
       }
-    }
+    },
+    blocks: []
   }
 
   return articleClient.create({
@@ -303,7 +304,8 @@ export const unpublishArticle = async (
               createMany: {
                 data: socialMediaAuthors.map(({authorId}) => ({authorId}))
               }
-            }
+            },
+            blocks: []
           },
           update: {
             ...revision,
@@ -333,7 +335,8 @@ export const unpublishArticle = async (
               createMany: {
                 data: socialMediaAuthors.map(({authorId}) => ({authorId}))
               }
-            }
+            },
+            blocks: []
           }
         }
       },
@@ -491,7 +494,8 @@ export const publishArticle = async (
                 createMany: {
                   data: socialMediaAuthors.map(({authorId}) => ({authorId}))
                 }
-              }
+              },
+              blocks: []
             },
             update: {
               ...revision,
@@ -521,7 +525,8 @@ export const publishArticle = async (
                 createMany: {
                   data: socialMediaAuthors.map(({authorId}) => ({authorId}))
                 }
-              }
+              },
+              blocks: []
             }
           }
         },
@@ -579,7 +584,8 @@ export const publishArticle = async (
               createMany: {
                 data: socialMediaAuthors.map(({authorId}) => ({authorId}))
               }
-            }
+            },
+            blocks: []
           },
           update: {
             ...revision,
@@ -609,7 +615,8 @@ export const publishArticle = async (
               createMany: {
                 data: socialMediaAuthors.map(({authorId}) => ({authorId}))
               }
-            }
+            },
+            blocks: []
           }
         }
       },
diff --git a/libs/api/src/lib/graphql/page.ts b/libs/api/src/lib/graphql/page.ts
index 56d380f0..6b63518e 100644
--- a/libs/api/src/lib/graphql/page.ts
+++ b/libs/api/src/lib/graphql/page.ts
@@ -210,7 +210,7 @@ export const GraphQLPublicPage = new GraphQLObjectType<PublicPage, Context>({
     properties: {
       type: GraphQLNonNull(GraphQLList(GraphQLNonNull(GraphQLMetadataPropertyPublic))),
       resolve: ({properties}) => {
-        return properties.filter(property => property.public).map(({key, value}) => ({key, value}))
+        return properties?.filter(property => property.public).map(({key, value}) => ({key, value}))
       }
     },
 
diff --git a/libs/api/src/lib/graphql/page/page.private-mutation.ts b/libs/api/src/lib/graphql/page/page.private-mutation.ts
index 8d21a995..456df84e 100644
--- a/libs/api/src/lib/graphql/page/page.private-mutation.ts
+++ b/libs/api/src/lib/graphql/page/page.private-mutation.ts
@@ -140,7 +140,8 @@ export const duplicatePage = async (
       createMany: {
         data: duplicatedProperties
       }
-    }
+    },
+    blocks: []
   }
 
   return pageClient.create({
@@ -236,7 +237,8 @@ export const unpublishPage = async (
               createMany: {
                 data: properties
               }
-            }
+            },
+            blocks: []
           },
           update: {
             ...revision,
@@ -250,7 +252,8 @@ export const unpublishPage = async (
               createMany: {
                 data: properties
               }
-            }
+            },
+            blocks: []
           }
         }
       },
@@ -387,7 +390,8 @@ export const publishPage = async (
                 createMany: {
                   data: properties
                 }
-              }
+              },
+              blocks: []
             },
             update: {
               ...revision,
@@ -401,7 +405,8 @@ export const publishPage = async (
                 createMany: {
                   data: properties
                 }
-              }
+              },
+              blocks: []
             }
           }
         },
@@ -443,7 +448,8 @@ export const publishPage = async (
               createMany: {
                 data: properties
               }
-            }
+            },
+            blocks: []
           },
           update: {
             ...revision,
@@ -454,7 +460,8 @@ export const publishPage = async (
               createMany: {
                 data: properties
               }
-            }
+            },
+            blocks: []
           }
         }
       },
