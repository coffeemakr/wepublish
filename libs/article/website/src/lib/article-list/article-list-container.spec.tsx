import {runStorybookContainerTests} from '@wepublish/testing'
import * as stories from './article-list-container.stories'

describe('AuthorList Container', () => {
  runStorybookContainerTests(stories)
})
