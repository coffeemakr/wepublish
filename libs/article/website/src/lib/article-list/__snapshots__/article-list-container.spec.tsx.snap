// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorList Container should render Default 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: grid;
  gap: 32px;
  grid-template-columns: repeat(1, 1fr);
}

@media (min-width:600px) {
  .emotion-0 {
    grid-template-columns: repeat(2, 1fr);
  }
}

@media (min-width:900px) {
  .emotion-0 {
    grid-template-columns: repeat(3, 1fr);
  }
}

@media (min-width:1200px) {
  .emotion-0 {
    grid-template-columns: repeat(4, 1fr);
  }
}

.emotion-2 {
  margin: 0;
  font: inherit;
  color: #1976d2;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  text-decoration-color: rgba(25, 118, 210, 0.4);
  display: grid;
  gap: 16px;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: flex-start;
  align-content: flex-start;
}

.emotion-2:hover {
  text-decoration-color: inherit;
}

.emotion-3 {
  display: grid;
  width: 100%;
}

.emotion-5 {
  max-width: 100%;
  width: 100%;
  height: auto;
  aspect-ratio: auto 0.6666666666666666;
  min-height: inherit;
  object-fit: cover;
}

.emotion-9 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1.5rem;
  line-height: 1.334;
  letter-spacing: 0em;
}

.emotion-10 {
  margin-top: 16px;
}

<article
    class="emotion-0 emotion-1"
  >
    <a
      class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways emotion-2"
      href="https://example.com"
    >
      <div
        class="emotion-3 emotion-4"
      >
        <img
          alt="DSC07717"
          class="emotion-5 emotion-6"
          loading="lazy"
          srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
          title=""
        />
      </div>
      <div
        class="emotion-7 emotion-8"
      >
        <time
          datetime="2023-01-01T00:00:00.000Z"
        >
          01.01.2023
        </time>
        <h1
          class="MuiTypography-root MuiTypography-h5 emotion-9"
        >
          title
        </h1>
        <div
          class="emotion-10 emotion-11"
        >
          Foobar, Foobar
        </div>
      </div>
    </a>
    <a
      class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways emotion-2"
      href="https://example.com"
    >
      <div
        class="emotion-3 emotion-4"
      >
        <img
          alt="DSC07717"
          class="emotion-5 emotion-6"
          loading="lazy"
          srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
          title=""
        />
      </div>
      <div
        class="emotion-7 emotion-8"
      >
        <time
          datetime="2023-01-01T00:00:00.000Z"
        >
          01.01.2023
        </time>
        <h1
          class="MuiTypography-root MuiTypography-h5 emotion-9"
        >
          title
        </h1>
        <div
          class="emotion-10 emotion-11"
        >
          Foobar, Foobar
        </div>
      </div>
    </a>
    <a
      class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways emotion-2"
      href="https://example.com"
    >
      <div
        class="emotion-3 emotion-4"
      >
        <img
          alt="DSC07717"
          class="emotion-5 emotion-6"
          loading="lazy"
          srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
          title=""
        />
      </div>
      <div
        class="emotion-7 emotion-8"
      >
        <time
          datetime="2023-01-01T00:00:00.000Z"
        >
          01.01.2023
        </time>
        <h1
          class="MuiTypography-root MuiTypography-h5 emotion-9"
        >
          title
        </h1>
        <div
          class="emotion-10 emotion-11"
        >
          Foobar, Foobar
        </div>
      </div>
    </a>
  </article>
</DocumentFragment>
`;

exports[`AuthorList Container should render WithClassName 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -1,8 +1,8 @@
  <DocumentFragment>
    <article
-     class="css-dp8k64-ArticleListWrapper ei1ca8h0"
+     class="extra-classname css-dp8k64-ArticleListWrapper ei1ca8h0"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1vgz1ax-MuiTypography-root-MuiLink-root-ArticleListItem"
        href="https://example.com"
      >"
`;

exports[`AuthorList Container should render WithEmotion 1`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;

exports[`AuthorList Container should render WithFilter 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -76,45 +76,7 @@
          >
            Foobar, Foobar
          </div>
        </div>
      </a>
-     <a
-       class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1vgz1ax-MuiTypography-root-MuiLink-root-ArticleListItem"
-       href="https://example.com"
-     >
-       <div
-         class="css-14c86dg-ArticleListItemImageWrapper e1tp74qn2"
-       >
-         <img
-           alt="DSC07717"
-           class="css-v2aane-ImageWrapper-ArticleListItem e1shv2yg0"
-           loading="lazy"
-           srcset="https://unsplash.it/800/800 800w,
- https://unsplash.it/500/500 500w,
- https://unsplash.it/300/300 300w,
- https://unsplash.it/200/200 200w"
-           title=""
-         />
-       </div>
-       <div
-         class="css-1w8pg9w-ArticleListItemContent e1tp74qn1"
-       >
-         <time
-           datetime="2023-01-01T00:00:00.000Z"
-         >
-           01.01.2023
-         </time>
-         <h1
-           class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
-         >
-           title
-         </h1>
-         <div
-           class="css-1o8pv0h-ArticleListItemAuthors e1tp74qn0"
-         >
-           Foobar, Foobar
-         </div>
-       </div>
-     </a>
    </article>
  </DocumentFragment>"
`;
